kyaml/yaml/match_test.go

```diff
@@ -74,11 +74,9 @@ spec:
 `},
 		{[]string{
 			"spec", "template", "spec", "containers", "[name=.*]", "ports"},
-			`-
-- containerPort: 80
--
-- containerPort: 8081
-- containerPort: 9090
+			`- - containerPort: 80
+- - containerPort: 8081
+  - containerPort: 9090
```

Comment:
@KnVerey It looks like [Donny's PR](https://github.com/Shell32-Natsu/yaml/commit/dd16c726352a7e6d641105100ec056a9f92fb7fd) solves the problem of nested lists, we can use what we need to from his code.